core/filtermaps/filtermaps.go

```diff
@@ -668,8 +668,12 @@ func (f *FilterMaps) mapRowIndex(mapIndex, rowIndex uint32) uint64 {
 // Note that this function assumes that the indexer read lock is being held when
 // called from outside the indexerLoop goroutine.
 func (f *FilterMaps) getBlockLvPointer(blockNumber uint64) (uint64, error) {
-	if blockNumber >= f.indexedRange.blocks.AfterLast() && f.indexedRange.headIndexed {
-		return f.indexedRange.headDelimiter + 1, nil
+	if blockNumber >= f.indexedRange.blocks.AfterLast() {
+		if f.indexedRange.headIndexed {
+			return f.indexedRange.headDelimiter + 1, nil
+		} else {
+			return uint64(f.indexedRange.maps.AfterLast()) << f.logValuesPerMap, nil
```

Comment:
Why do we return the starting position of log value of the next map, if the current chain is not fully indexed?

Reply:
Indeed it makes more sense to just return the start pointer of the last block.